linkerd/telemeter/usage/src/main/scala/io/buoyant/linkerd/telemeter/UsageDataTelemeter.scala

```diff
@@ -0,0 +1,201 @@
+    )
+
+  val RequestPattern = s"""^*/srv/*/requests$$""".r
+  def mkCounters(metrics: Map[String, Number]): Seq[Counter] =
+    metrics.filter({
+      case (RequestPattern(key), value) => true
+      case _ => false
+    }).map({
+      case (k, v: Number) => Counter(Some("srv_requests"), Some(v.longValue()))
+    }).toSeq
+
+  def mkRouters(config: Linker.LinkerConfig): Seq[Router] =
+    config.routers.map(r => {
+
+      val identifiers = r match {
+        case httpRouter: HttpConfig =>
+          httpRouter.identifier.getOrElse(Seq()).map(_.kind)
+        case _ => Seq()
+      }
+
+      val transformers = r.interpreter.transformers.getOrElse(Seq()).map(_.kind)
+
+      Router(
+        protocol = Some(r.protocol.configId),
+        interpreter = Some(r.interpreter.kind),
+        identifiers = identifiers,
+        transformers = transformers
+      )
+    })
+
+  class UsageDataHandler(
+    service: Service[Request, Response],
+    config: Linker.LinkerConfig,
+    orgId: Option[String],
+    registry: Metrics
+  ) extends Admin.Handler {
+
+    def apply(request: Request): Future[Response] = {
+      val msg: UsageMessage = mkUsageMessage(config, orgId, registry.sample().asScala.toMap)
+      val rsp = Response()
+      rsp.contentType = "application/json"
+      rsp.contentString = mapper.writeValueAsString(msg)
+      Future.value(rsp)
+    }
+  }
+
+  val mapper = new ObjectMapper(new JsonFactory) with ScalaObjectMapper
+  mapper.registerModule(DefaultScalaModule)
+  mapper.setSerializationInclusion(Include.NON_ABSENT)
+  mapper.setVisibility(PropertyAccessor.ALL, Visibility.PUBLIC_ONLY)
+}
+
+class UsageDataTelemeter(
+  metricsDst: Name,
+  config: Linker.LinkerConfig,
+  registry: com.twitter.common.metrics.Metrics,
+  orgId: Option[String]
+) extends Telemeter with Admin.WithHandlers {
+  import UsageDataTelemeter._
+
+  val tracer = NullTracer
+  val stats = NullStatsReceiver
+
+  private[this] val metricsService = Http.client.newService(metricsDst, "usageData")
+  private[this] val started = new AtomicBoolean(false)
+
+  val adminHandlers: Admin.Handlers = Seq(
+    "/admin/metrics/usage" -> new UsageDataHandler(metricsService, config, orgId, registry)
+  )
+
+  def sample(registry: Metrics) = registry.sample().asScala.toMap
+
+  // Only run at most once.
+  def run(): Closable with Awaitable[Unit] =
+    if (started.compareAndSet(false, true)) run0()
+    else Telemeter.nopRun
+
+  private[this] def run0() = {
+    val client = Client(metricsService, config, orgId)
+    val _ = client(sample(registry))
+
+    val task = DefaultTimer.twitter.schedule(DefaultPeriod) {
+      val _ = client(sample(registry))
+    }
+
+    val closer = Closable.sequence(
+      task,
+      Closable.all(metricsService)
+    )
+
+    new Closable with CloseAwaitably {
+      def close(deadline: Time) = closeAwaitably {
+        closer.close(deadline)
+      }
+    }
+  }
+}
+
+case class UsageDataTelemeterConfig(
+  orgId: Option[String]
```

Comment:
No, only trying to be as transparent as possible about what's being sent. Currently in order to use `/admin/metrics/usage` to see what's being sent, the telemeter needs to be loaded. But the telemeter also sends an initial payload on load. If we think docs + protobuf file sets enough context then this may not be an issue.